### DISCUSSION TITLE
Supertest wrapper for API testing

### DISCUSSION THREAD

**Discussion by epotter2297** [ecyrbe/zodios#306] [Ideas]
Right now, when I run the test suite for my backend (using `@zodios/core` and `@zodios/express`), I start the express app and query it in each test by using the zodios client. Having a type-safe way to send requests and get responses in my test files is very nice, but I lose out on many quality of life features that I would have if I stuck with supertest, like being able to do `supertest(app).get('/thing/incorrectId').expect(404)`.

In the future, it would be awesome if zodios provided a method of wrapping supertest to add type-safety, since it would be the best of both worlds 🙂

**Comment by ecyrbe** [comment]
So zodios v11 has a testing libray in the works :  https://github.com/ecyrbe/zodios/tree/feature/v11/packages/testing
But for now it's the other way around and it's a mocking library like moxios to mock your backend.
A complement would be a api fetcher with testing hability.
Like 
```ts
const client = new Zodios(...);
const zodiostest = new ZodiosTest(client, app);

zodiosTest.get('/thing/:id', { params: incorrectId }).expect(404);
```

So this could be possible, but not for v11, as i'm already out of schedule.
But if someone want to help implement this. i'll be glad to help.
